router/router.nginx.conf.in

```diff
@@ -232,3 +232,20 @@ server {
     listen 80;
     listen [::]:80;
 }
+
+server {
+    server_name blog.*;
+
+    location / {
+        proxy_pass http://blog/;
+
+        proxy_set_header Host $updated_host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto https;
```

Comment:
I think this would be slightly cleaner as `$updated_scheme` instead of `https`, so that we can avoid hardcoding protocols (not a big deal here since we will likely never want anything besides https). $updated_scheme is slightly confusingly named, but represents the passed-through $scheme (protocol) from gateway ($scheme is the current request protocol, which will be https for both http and https requests, since http requests are always redirected to https).


Ghost also recommends you set the X-Forwarded-Host header: you can do so by writing `proxy_set_header X-Forwarded-Host $updated_host`. I don't think you will have a problem until we try to run ghost behind internal.hail.is (for dev purposes), because gateway sets the proxied request's HOST to $http_host when running on hail.is (but for internal.hail.is sets the Host to $service.internal where service comes from internal.hail.is/namespace/{service}). I'm assuming ghost defaults to the Host when X-Forwarded-Host isn't set.

https://ghost.org/faq/change-configured-site-url/#setup-ssl